frontend/dockerfile/instructions/parse_test.go

```diff
@@ -187,16 +188,21 @@ func TestErrorCases(t *testing.T) {
       bar`,
 			expectedError: "COPY requires at least two arguments",
 		},
+		{
+			name:          "COPY heredoc destination",
+			dockerfile:    "COPY /foo <<EOF\nEOF",
+			expectedError: "COPY cannot accept a heredoc as a destination",
+		},
+		{
+			name:          "COPY heredoc extra args",
+			dockerfile:    "COPY <<EOF /foo /bar\nEOF",
+			expectedError: "COPY requires exactly two arguments",
```

Comment:
There was a comment proposing to allow the here document to have the "content" and copy it to a "destination" https://github.com/moby/moby/issues/34423#issuecomment-321068115

```dockerfile
# write "this is contents" to a file named /dest
COPY  <<EOF /dest
this is contents
EOF
```

It's probably ok to not implement everything in the first iteration, but we need to be sure that the syntax / implementation allows for these things to be added in future

Reply:
Yup, under the current state of the codebase that snippet would work! However, the parser currently rejects the flipped case:

```docker
COPY /src <<EOF
this is contents
EOF
```

Writing *to* a heredoc doesn't make much sense to me, so I have the parser reject these cases.

I'd really like to get the ADD/COPY syntax like this in the first iteration, I really like it for expressing configs inline :tada: